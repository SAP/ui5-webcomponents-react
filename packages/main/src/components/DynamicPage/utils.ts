export const DynamicPageCssVariables = {
  headerDisplay: '--_ui5wcr_DynamicPage_header_display',
  titleFontSize: '--_ui5wcr_DynamicPage_title_fontsize'
};
