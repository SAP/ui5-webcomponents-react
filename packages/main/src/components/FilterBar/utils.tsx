import { cloneElement } from 'react';

const inputTagNames = new Set([
  'UI5-COMBOBOX',
  'UI5-DATE-PICKER',
  'UI5-DATERANGE-PICKER',
  'UI5-DATETIME-PICKER',
  'UI5-DURATION-PICKER',
  'UI5-INPUT',
  'UI5-MULTI-INPUT',
  'UI5-RATING-INDICATOR',
  'UI5-STEP-INPUT',
  'UI5-TIME-PICKER'
]);

export const filterValue = (ref, child) => {
  const tagName = ref.tagName;
  let filterItemProps = {};
  if (inputTagNames.has(tagName)) {
    filterItemProps = { value: ref.value };
  }
  if (tagName === 'UI5-SELECT' || tagName === 'UI5-MULTI-COMBOBOX') {
    const selectedIndices = Array.from(ref.children as HTMLCollectionOf<any>)
      .map((item, index) => (item.selected ? index : false))
      .filter((el) => el !== false);
    const selectedIndicesSet = new Set(selectedIndices);
    const options = child.props.children.props.children?.map((item, index) => {
      if (selectedIndicesSet.has(index)) {
        return cloneElement(item, { selected: true });
      }
      return cloneElement(item, { selected: false });
    });
    filterItemProps = { children: options };
  }
  if (tagName === 'UI5-SWITCH' || tagName === 'UI5-CHECKBOX') {
    filterItemProps = { checked: ref.checked };
  }
  return filterItemProps;
};

export const renderSearchWithValue = (renderSearchElement, searchValue) => {
  return cloneElement(renderSearchElement, { value: searchValue ?? renderSearchElement?.props?.value });
};
