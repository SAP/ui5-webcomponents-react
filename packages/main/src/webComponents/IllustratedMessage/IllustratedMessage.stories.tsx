import { Meta, StoryObj } from '@storybook/react';
import '@ui5/webcomponents-fiori/dist/illustrations/AllIllustrations.js';
import { IllustrationMessageType } from '../../enums/IllustrationMessageType';
import { IllustratedMessage } from './index';

const meta = {
  title: 'User Feedback / IllustratedMessage',
  component: IllustratedMessage,
  args: {
    name: IllustrationMessageType.BeforeSearch
  },
  argTypes: {
    children: { control: { disable: true } }
  }
} satisfies Meta<typeof IllustratedMessage>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
