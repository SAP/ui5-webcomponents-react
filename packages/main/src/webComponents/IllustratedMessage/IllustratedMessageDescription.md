## Structure

The IllustratedMessage consists of the following elements, which are displayed below each other in the following order:  

*   Illustration
*   Title
*   Subtitle
*   Actions

## Usage

`IllustratedMessage` is meant to be used inside container component, for example a `Card`, a `Dialog` or a `Page`