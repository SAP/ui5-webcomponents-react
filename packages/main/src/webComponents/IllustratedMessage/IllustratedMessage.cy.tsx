import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { IllustratedMessage } from './index';

describe('IllustratedMessage', () => {
  it('Basic Test (generated)', () => {
    cy.mount(<IllustratedMessage />);
    cy.get('[ui5-illustrated-message]').should('exist');
  });

  cypressPassThroughTestsFactory(IllustratedMessage);
});
