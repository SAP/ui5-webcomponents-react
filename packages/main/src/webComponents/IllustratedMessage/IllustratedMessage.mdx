import { ControlsWithNote, DocsHeader, Footer } from '@sb/components';
import { Canvas, Markdown, Meta } from '@storybook/blocks';
import { MessageStripDesign } from '../../enums';
import { MessageStrip } from '../MessageStrip';
import * as ComponentStories from './IllustratedMessage.stories';
import Description from './IllustratedMessageDescription.md?raw';
import { IllustratedMessage } from './index';

<Meta of={ComponentStories} />

<DocsHeader since="0.18.0" />

<br />

## Example

<Canvas of={ComponentStories.Default} story={{ height: '300px' }} layout={'fullscreen'} />

## Properties

<ControlsWithNote of={ComponentStories.Default} />

<Markdown>{Description}</Markdown>

## Fully customizable title

<MessageStrip
  design={MessageStripDesign.Warning}
  hideCloseButton
  children={
    <>
      As the <code>title</code> slot name collides with the standard HTML <code>title</code> attribute, using the{' '}
      <code>title</code> prop will not work. <br />
      You must pass the <code>title</code> slot as part of the components <code>children</code>.
    </>
  }
/>

In some cases you might need to fully customize the title element of the `IllustratedMessage`.
You can achieve this by using the `title` slot.

**Note:** Using this slot, the default title text of illustration and the value of `titleText` property will be overwritten.

```jsx
<IllustratedMessage>
  <h5 slot="title">This is my fully customized title</h5>
</IllustratedMessage>
```

<IllustratedMessage>
  <h5 slot="title">This is my fully customized title</h5>
</IllustratedMessage>

<Footer />
