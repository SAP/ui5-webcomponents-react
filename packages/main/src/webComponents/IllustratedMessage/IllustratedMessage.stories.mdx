import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs';
import { IllustratedMessage } from '@ui5/webcomponents-react/dist/IllustratedMessage';
import { IllustrationMessageType } from '@ui5/webcomponents-react/dist/IllustrationMessageType';
import { DocsHeader } from '@shared/stories/DocsHeader';
import '@ui5/webcomponents-fiori/dist/illustrations/BeforeSearch';
import '@ui5/webcomponents-fiori/dist/illustrations/NoActivities';
import '@ui5/webcomponents-fiori/dist/illustrations/NoData';
import '@ui5/webcomponents-fiori/dist/illustrations/NoEntries';
import '@ui5/webcomponents-fiori/dist/illustrations/NoMail';
import '@ui5/webcomponents-fiori/dist/illustrations/NoNotifications';
import '@ui5/webcomponents-fiori/dist/illustrations/NoSavedItems';
import '@ui5/webcomponents-fiori/dist/illustrations/NoSearchResults';
import '@ui5/webcomponents-fiori/dist/illustrations/NoTasks';
import '@ui5/webcomponents-fiori/dist/illustrations/UnableToLoad';
import '@ui5/webcomponents-fiori/dist/illustrations/UnableToUpload';

<Meta
  title="User Feedback / IllustratedMessage"
  component={IllustratedMessage}
  argTypes={{
    children: { control: { disable: true } }
  }}
  args={{
    name: IllustrationMessageType.BeforeSearch
  }}
/>

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return <IllustratedMessage {...args} />;
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />

## Usage

`IllustratedMessage` is meant to be used inside container component, for example a `Card`, a `Dialog` or a `Page`.
