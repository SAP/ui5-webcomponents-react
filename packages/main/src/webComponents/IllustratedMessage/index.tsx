import { ReactNode } from 'react';
import { IllustrationMessageType } from '../../enums';
import { CommonProps } from '../../interfaces/CommonProps';
import { Ui5DomRef } from '../../interfaces/Ui5DomRef';
import { withWebComponent } from '../../internal/withWebComponent';

import '@ui5/webcomponents-fiori/dist/IllustratedMessage.js';

interface IllustratedMessageAttributes {
  /**
   * Defines the illustration name that will be displayed in the component.
   *
   * Available illustrations are:
   *
   * *   `BeforeSearch`
   * *   `NoActivities`
   * *   `NoData`
   * *   `NoMail`
   * *   `NoMail_v1`
   * *   `NoEntries`
   * *   `NoNotifications`
   * *   `NoSavedItems`
   * *   `NoSavedItems_v1`
   * *   `NoSearchResults`
   * *   `NoTasks`
   * *   `NoTasks_v1`
   * *   `UnableToLoad`
   * *   `UnableToLoadImage`
   * *   `UnableToUpload`
   * *   `AddColumn`
   * *   `AddPeople`
   * *   `BalloonSky`
   * *   `Connection`
   * *   `EmptyCalendar`
   * *   `EmptyList`
   * *   `EmptyPlanningCalendar`
   * *   `ErrorScreen`
   * *   `FilterTable`
   * *   `GroupTable`
   * *   `NoFilterResults`
   * *   `PageNotFound`
   * *   `ReloadScreen`
   * *   `ResizeColumn`
   * *   `SearchEarth`
   * *   `SearchFolder`
   * *   `SimpleBalloon`
   * *   `SimpleBell`
   * *   `SimpleCalendar`
   * *   `SimpleCheckMark`
   * *   `SimpleConnection`
   * *   `SimpleEmptyDoc`
   * *   `SimpleEmptyList`
   * *   `SimpleError`
   * *   `SimpleMagnifier`
   * *   `SimpleMail`
   * *   `SimpleNoSavedItems`
   * *   `SimpleNotFoundMagnifier`
   * *   `SimpleReload`
   * *   `SimpleTask`
   * *   `SleepingBell`
   * *   `SortColumn`
   * *   `SuccessBalloon`
   * *   `SuccessCheckMark`
   * *   `SuccessHighFive`
   * *   `SuccessScreen`
   * *   `Tent`
   * *   `UploadCollection`
   * *   `TntCompany`
   * *   `TntExternalLink`
   * *   `TntFaceID`
   * *   `TntFingerprint`
   * *   `TntLock`
   * *   `TntMission`
   * *   `TntNoApplications`
   * *   `TntNoFlows`
   * *   `TntNoUsers`
   * *   `TntRadar`
   * *   `TntServices`
   * *   `TntSessionExpired`
   * *   `TntSessionExpiring`
   * *   `TntSuccess`
   * *   `TntSuccessfulAuth`
   * *   `TntUnlock`
   * *   `TntUnsuccessfulAuth`
   *
   *
   *
   * **Note:** By default the `BeforeSearch` illustration is loaded.
   * When using an illustration type, other than the default, it should be loaded in addition:
   * `import "@ui5/webcomponents-fiori/dist/illustrations/NoData.js";`
   *
   * **Note:** TNT illustrations cointain `Tnt` prefix in their name. You can import them removing the `Tnt` prefix like this:
   * `import "@ui5/webcomponents-fiori/dist/illustrations/tnt/SessionExpired.js";`
   */
  name?: IllustrationMessageType | keyof typeof IllustrationMessageType;
  /**
   * Defines the subtitle of the component.
   *
   * **Note:** Using this property, the default subtitle text of illustration will be overwritten.
   *
   * **Note:** Using `subtitle` slot, the default of this property will be overwritten.
   */
  subtitleText?: string;
  /**
   * Defines the title of the component.
   *
   * **Note:** Using this property, the default title text of illustration will be overwritten.
   */
  titleText?: string;
}

export interface IllustratedMessageDomRef extends IllustratedMessageAttributes, Ui5DomRef {}

export interface IllustratedMessagePropTypes extends IllustratedMessageAttributes, CommonProps {
  /**
   * Defines the component actions.
   */
  children?: ReactNode | ReactNode[];
  /**
   * Defines the subtitle of the component.
   *
   * **Note:** Using this slot, the default subtitle text of illustration and the value of `subtitleText` property will be overwritten.
   *
   * __Note:__ When passing a custom React component to this prop, you have to make sure your component reads the `slot` prop and appends it to the most outer element of your component.
   * Learn more about it [here](https://sap.github.io/ui5-webcomponents-react/?path=/docs/knowledge-base--page#adding-custom-components-to-slots).
   */
  subtitle?: ReactNode;
}

/**
 * An IllustratedMessage is a recommended combination of a solution-oriented message, an engaging illustration, and conversational tone to better communicate an empty or a success state than just show a message alone. Each illustration has default internationalised title and subtitle texts. Also they can be managed with `titleText` and `subtitleText` properties. To display the desired illustration, use the `name` property, where you can find the list of all available illustrations.
 *
 * **Note:** By default the “BeforeSearch” illustration is loaded. To use other illustrations, make sure you import them in addition, for example:
 * `import "@ui5/webcomponents-fiori/dist/illustrations/NoData.js"`
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/IllustratedMessage" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const IllustratedMessage = withWebComponent<IllustratedMessagePropTypes, IllustratedMessageDomRef>(
  'ui5-illustrated-message',
  ['name', 'subtitleText', 'titleText'],
  [],
  ['subtitle'],
  []
);

IllustratedMessage.displayName = 'IllustratedMessage';

IllustratedMessage.defaultProps = {
  name: IllustrationMessageType.BeforeSearch
};

export { IllustratedMessage };
