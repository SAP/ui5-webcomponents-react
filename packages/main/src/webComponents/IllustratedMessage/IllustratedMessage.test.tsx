import { render } from '@shared/tests';
import { createCustomPropsTest } from '@shared/tests/utils';
import { IllustratedMessage } from '@ui5/webcomponents-react/dist/IllustratedMessage';
import React from 'react';

describe('IllustratedMessage', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(<IllustratedMessage />);
    expect(asFragment()).toMatchSnapshot();
  });
  createCustomPropsTest(IllustratedMessage);
});
