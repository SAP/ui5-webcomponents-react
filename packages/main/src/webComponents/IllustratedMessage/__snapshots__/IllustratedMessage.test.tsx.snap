// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IllustratedMessage Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-illustrated-message
    name="BeforeSearch"
    ui5-illustrated-message=""
  />
</DocumentFragment>
`;
